Marlin/cardreader.cpp

```diff
@@ -83,7 +83,8 @@ void CardReader::lsDive(const char *prepend, SdFile parent, const char * const m
       // Allocate enough stack space for the full path to a folder, trailing slash, and nul
       boolean prepend_is_empty = (prepend[0] == '\0');
       int len = (prepend_is_empty ? 1 : strlen(prepend)) + strlen(lfilename) + 1 + 1;
-      char path[len];
+      char *path=(char*)malloc(len);
```

Comment:
Whether the use of malloc() is a good or bad thing in embedded programs, is a philosophic question we answered for Marlin with _NO_!